model/Bom.js

```diff
@@ -38,11 +41,47 @@ class Bom extends CycloneDXObject {
     if (pkg) {
       this._metadata = this.createMetadata(pkg, componentType)
       this._components = this.listComponents(pkg, lockfile)
+      this._dependencies = this.listDependencies(pkg)
     } else {
       this._components = []
+      this._dependencies = []
     }
   }
 
+  listDependencies (pkg) {
+    const list = []
+    this.createDependency(pkg, list)
+    return list
+  }
+
+  createDependency (pkg, list) {
+    // read-installed with default options marks devDependencies as extraneous
+    // if a package is marked as extraneous, do not include it as a dependency
+    // if -d or --include-dev is specified then all dev dependencies are considered
+    if (pkg.extraneous) return
+    const ref = this.createPackageURL(pkg)
+    const deplist = []
+    if (pkg._dependencies && Object.keys(pkg._dependencies).length !== 0) {
+      Object.keys(pkg._dependencies)
+        .map(x => pkg.dependencies[x])
+        .filter(x => x !== undefined && typeof (x) !== 'string') // remove cycles
+        .map(x => deplist.push(new Dependency(this.createPackageURL(x), this.createDependency(x, list))))
+      list.push(new Dependency(ref, deplist))
+    }
+    return deplist
+  }
+
+  createPackageURL (pkg) {
```

Comment:
:exclamation: there is code to crate a PURL already. see https://github.com/CycloneDX/cyclonedx-node-module/blob/b8f6940a4b7228e048a5d81b8115c1b8a6cf0f7d/model/Component.js#L43

if you really want to use a PURL as the `bom-ref`'s value, please incorporate the already existing PURL creation.
Reasons: if a `bom-ref` looks like a PURL, i would expect it to be the actual PURL of a `Component`, not a string that was crafted somehow else.

Reply:
This is the reason I wanted to implement the BomReference class and make use of creation of PURL in both Component and Bom. Is it okay to do something like this?

Reply:
@jkowalleck Any suggestions on this?

Reply:
see https://github.com/CycloneDX/cyclonedx-node-module/pull/244#issuecomment-1024994973